src/libstd/ffi/c_str.rs

```diff
@@ -1143,7 +1140,13 @@ impl CStr {
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn to_bytes_with_nul(&self) -> &[u8] {
-        unsafe { &*(&self.inner as *const [c_char] as *const [u8]) }
+        // safety: safe references to `CStr` can only exist if they point to a memory location
+        // that is terminated by a `\0`.
```

Comment:
Is there a reason the original code didn't assert that the pointer was not null? I'm guessing it either caused miscompilation or pessmizations.

Reply:
I don't know. I'm guessing an oversight?